pkg/tnf/handlers/handler_template/handler.tmpl

```diff
@@ -0,0 +1,96 @@
+// Copyright (C) 2020-2021 Red Hat, Inc.
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+package {{ .Handlername }}
+
+import (
+)
+//TODO
+// Hostname provides a hostname test implemented using command line tool "hostname".
+/*
+type {{ .Handlername }} struct {
+	result  int
+	timeout time.Duration
+	args    []string
+	// The hostname
+	hostname string
+}
+
+const (
+	// Command is the command name for the unix "hostname" command.
+	Command = {{ .Handlername }}
+	// SuccessfulOutputRegex is the regular expression match for hostname output.
+	SuccessfulOutputRegex = `.+`
+)
+
+// Args returns the command line args for the test.
+func (h *{{ .Handlername }}) Args() []string {
+	return h.args
+}
+
+// GetIdentifier returns the tnf.Test specific identifier.
+func (h *{{ .Handlername }}) GetIdentifier() identifier.Identifier {
+	return identifier.HostnameIdentifier
+}
+
+// Timeout return the timeout for the test.
+func (h *{{ .Handlername }}) Timeout() time.Duration {
+	return h.timeout
+}
+
+// Result returns the test result.
+func (h *{{ .Handlername }}) Result() int {
+	return h.result
+}
+
+// ReelFirst returns a step which expects an hostname summary for the given device.
+func (h *{{ .Handlername }}) ReelFirst() *reel.Step {
+	return &reel.Step{
+		Expect:  []string{SuccessfulOutputRegex},
+		Timeout: h.timeout,
+	}
+}
+
+// ReelMatch parses the hostname output and set the test result on match.
+// Returns no step; the test is complete.
+func (h *{{ .Handlername }}) ReelMatch(_, _, match string) *reel.Step {
+	h.hostname = match
+	h.result = tnf.SUCCESS
+	return nil
+}
+
+// ReelTimeout does nothing;  hostname requires no explicit intervention for a timeout.
+func (h *{{ .Handlername }}) ReelTimeout() *reel.Step {
```

Comment:
there's few places where hostname is still mentionned.
The code still needs more cleaning to remove out non-generic parts and replace them with Action/TODO..